### PR TITLE
Prefer $SPACEMACDIR/init.el over ~/.spacemacs

### LINKED ISSUE #6669: Wrong init file when using SPACEMACSDIR and ~/.spacemacs exists
I'm trying to make Emacs (on a Windows machine) work with a Spacemacs installation (.emacs.d) and configuration (.spacemacs.d) that are stored on a USB flash drive. If I have a `~/.spacemacs`, and I use `SPACEMACSDIR` environment variable to specify a directory on the USB drive, then Spacemacs tries to load `~/.spacemacs` instead of `%SPACEMACSDIR%/init.el`. The reason is that `dotspacemacs-filepath` is set wrong. PR incoming. There are no repro steps because I'm in the middle of trying to get it working.

### PR DESCRIPTION
**Pull request by bmag**
Previously, if SPACMEACSDIR is used, and both `$SPACEMACDIR/init.el` and `~/.spacemacs` exist, Spacemacs would use `~/.spacemacs` as the dotspacemacs file. Obviously, SPACEMACSDIR is for using a different config than the one in the home directory, so the correct thing to do is use `$SPACEMACSDIR/init.el`. Fixes #6669.

I am not sure what to do when SPACEMACSDIR is specified but `$SPACEMACSDIR/init.el` doesn't exist. For now I decided to fallback to `~/.spacemacs`, but I suspect the better option is to set `dotspacemacs-filepath` to `$SPACMEACSDIR/init.el` so that Spacemacs installs a new dotfile there.

### CODE REVIEW

**Comment by d12frosted** [comment]
Oh, gonna copy-paste my comment here.

---

> but I suspect the better option is to set dotspacemacs-filepath to $SPACMEACSDIR/init.el so that Spacemacs installs a new dotfile there.

Ya, I think it would be better.

**Comment by bmag** [comment]
@d12frosted changed to use and install `$SPACEMACSDIR/init.el`also  when the file doesn't exist.
Will squash when there's a final decision.

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop. You can safely delete your branch.

**Comment by dijonkitchen** [comment]
Why are these cherry-picked instead of merged into the branch? 

Also, what's the cherry-picked commit so that we can check if it's in master yet or not?

**Comment by bmag** [comment]
> Why are these cherry-picked instead of merged into the branch?

To keep the history linear, mostly.

> what's the cherry-picked commit so that we can check if it's in master yet or not?

Don't know the commit ID, but you can search by commit message. Anyway, this commit is in master.

**Comment by dijonkitchen** [comment]
Thanks, I found it here: https://github.com/syl20bnr/spacemacs/commit/6d60d2ef05869ca6007f72befdfd7c7aadd49c50

I figure [rebasing](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase) to keep history linear would be easier than cherry-picking commits though.

**Comment by d12frosted** [comment]
TL;DR it's a matter of preference 😸 

@dijonkitchen rebase (in terms of dealing with PR) is usually made on the feature branch and then you merge the rebased feature branch into master which equals to cherry-picking commits. Equals in the sense of the result. You have no way to tell the difference just by looking at the commit in the main branch (be it `master`, `develop` or whatever). Also, when the feature branch has only one commit it's usually simpler to use cherry-pick (you can simply cherry-pick the 'branch').

I don't know about the workflow of the others, but personally, I don't use GitHub buttons for merging PRs (I know that it can squash/rebase and all the sweet stuff) - I deal with PRs in Emacs. Sometimes I cherry-pick them into `develop` branch and use them a while before actually pushing. AFAIK, Sylvain uses a similar approach, but he even marks PRs that he cherry-picked locally with a 'Merging...' label.

**Comment by d12frosted** [comment]
> what's the cherry-picked commit so that we can check if it's in master yet or not?

BTW, even if you use rebase-merge approach, the commit hash will change, because it most likely will have a different parent.

> You have no way to tell the difference just by looking at the commit in the main branch

Ok, I just realised that one of the side effects of using cherry-pick is that the cherry-picked commit is committed by the person performing cherry-pick. 

But when you rebase-merge PR that is already up to date with the main/target branch, committer is the author of PR, not the person doing rebase-merge. 

I hope I didn't miss something :)

**Comment by dijonkitchen** [comment]
Thanks for taking the time to answer so clearly. It's definitely a matter of preference (and perhaps ease with Emacs); I figure the transparency for other users would be good on Github though so people can track things and what's where with little effort. Maybe the best approach if you all want to stick with what you're doing is to just mention the new commit hash in a comment for posterity. 

Thanks again for all your great work! Love Spacemacs!

**Comment by d12frosted** [comment]
> Thanks for taking the time to answer so clearly

No problem :) 

> Maybe the best approach if you all want to stick with what you're doing is to just mention the new commit hash in a comment for posterity.

Good idea. Maybe I'll start doing this and we'll see how it's going.
